src/solver/conv_hip_implicit_gemm_3d_grouped_fwd_xdlops.cpp

```diff
@@ -0,0 +1,477 @@
+                                            const AnyInvokeParams& invoke_ctx) const
+{
+    return GenericSearch(*this, ctx, problem, invoke_ctx);
+}
+
+bool ConvHipImplicitGemm3DGroupFwdXdlops::IsApplicable(const ConvolutionContext& ctx,
+                                                       const ProblemDescription& problem) const
+{
+#if !MIOPEN_BACKEND_HIP || !MIOPEN_USE_COMPOSABLEKERNEL
+    std::ignore = ctx;
+    std::ignore = problem;
+    return false;
+#else
+    if(miopen::IsDisabled(MIOPEN_DEBUG_3D_CONV_IMPLICIT_GEMM_HIP_FWD_XDLOPS{}))
+        return false;
+    if(miopen::IsEnabled(MIOPEN_DEBUG_CONVOLUTION_DETERMINISTIC{}))
+        return false;
+    if(problem.conv_problem.GetInDataType() != problem.conv_problem.GetWeightsDataType() ||
+       problem.conv_problem.GetWeightsDataType() != problem.conv_problem.GetOutDataType() ||
+       problem.conv_problem.GetInDataType() != problem.conv_problem.GetOutDataType())
+        return false;
+    if(!problem.direction.IsForward())
+        return false;
+    if(!problem.Is3d())
+        return false;
+    if(!problem.IsLayoutNHWC())
+        return false;
+    const std::string& arch = ctx.GetStream().GetDeviceName();
+    if(!(arch == "gfx908" || arch == "gfx90a"))
+        return false;
+    switch(problem.conv_problem.GetInDataType())
+    {
+    case miopenHalf: return CheckCKApplicability<ck::half_t>(problem);
+    case miopenFloat: return CheckCKApplicability<float>(problem);
+    case miopenInt8: return CheckCKApplicability<int8_t>(problem);
+    case miopenInt32:
+    case miopenInt8x4:
+    case miopenBFloat16:
+    case miopenDouble: break;
+    }
+    return false;
+#endif
+}
+
+ConvSolution ConvHipImplicitGemm3DGroupFwdXdlops::GetSolution(
+    const ConvolutionContext& ctx,
+    const ProblemDescription& problem,
+    const PerformanceConfigHipImplicitGemm3DGroupFwdXdlops& config) const
+{
+    std::ignore = ctx;
+#if !MIOPEN_BACKEND_HIP || !MIOPEN_USE_COMPOSABLEKERNEL
+    std::ignore = problem;
+    std::ignore = config;
+    return {};
+#else
+    ConvSolution result;
+    result.invoker_factory = [=](const std::vector<Kernel>& kernels) {
+        std::ignore = kernels;
+        return [=](const Handle& handle, const AnyInvokeParams& primitive_parameters) {
+            switch(problem.conv_problem.GetInDataType())
+            {
+            case miopenHalf:
+                RunCKSolution<ck::half_t>(handle, primitive_parameters, problem, config);
+                break;
+            case miopenFloat:
+                RunCKSolution<float>(handle, primitive_parameters, problem, config);
+                break;
+            case miopenInt8:
+                RunCKSolution<int8_t>(handle, primitive_parameters, problem, config);
+                break;
+            case miopenInt32:
+            case miopenInt8x4:
+            case miopenBFloat16:
+            case miopenDouble: break;
+            }
```

Comment:
[Performance] The most of these computations can (and should) be moved to the invoker **_factory_**. Again, the library creates a dedicated instance of invoker for each specific problem. In the invoker's context, the problem is **_constant_**. Therefore parsing its values and building parameters in the invoker's context is a waste of CPU time.

Reply:
@atamazov  I will refactor the code as suggested. Thanks.

Reply:
@iq136boy Original comment updated: added notice that computations can be moved even higher, into `GetSolution`.

Reply:
@iq136boy Unfortunately, the code basically remains the same. Let me clarify: moving the code into a function does not change anything (and in this case highly recommended to be reverted unless you want to call the function from two or more places in the code). `Invoker` is an inner lambda, and what we need to do is move computations out from it. AFAICS the most of excessive computations are performed in RunCKSolution.

Please refer at #2305 -- this PR is intended to resolve the same problem. Hope this helps.

Reply:
@atamazov Thanks for the [clarification](https://github.com/ROCmSoftwarePlatform/MIOpen/pull/2286#discussion_r1291779229)
I have a quick fix to move parts of the computation from invoker. For more refactoring work to improve performance will be addressed in a separate PR.